src/PostgREST/Error.hs

```diff
@@ -73,6 +74,10 @@ binaryFieldError =
   simpleError HT.status406 (toS (toMime CTOctetStream) <>
   " requested but a single column was not selected")
 
+connectionLostError :: Response
```

Comment:
A special message for the reconnection state is needed. I tried to make your suggestion work by adding a new constructor for ```ApiRequestError``` with this message, all good, the problem was that by making the ```IORef (Either ApiRequestError DbStructure)``` I had to change two function type signatures, namely ```app``` and ```transactionMode```, to avoid that I could have unwrapped the Left of the IORef, but that would be the same non composable way as of the Maybe. In the end I was not able to make the code simpler with your suggestion.